src/ape/managers/query.py

```diff
@@ -28,9 +28,8 @@ def estimate_block_query(self, query: BlockQuery) -> Optional[int]:
 
     @estimate_query.register
     def estimate_block_transaction_query(self, query: BlockTransactionQuery) -> int:
-        block = self.provider.get_block(query.block_id)
-        # NOTE: Very loose estimate of 100ms per transaction in block for this query.
-        return len(block.transactions) * 100
+        # NOTE: Very loose estimate of 1000ms per block for this query.
+        return 1000
```

Comment:
This only makes a single call to ProviderAPI?

Reply:
@fubuloubu that causes ape to hang, and then it fails. I was going to make an issue for that

Reply:
Oh, it might have a segfault from recursion... hmm

Try this instead!
```suggestion
        return block.num_transactions * 100
```

Reply:
[db601da](https://github.com/ApeWorX/ape/pull/680/commits/db601da7ecac843d89bebc1b18764f40eb6e220e)